### PR TITLE
Feat/node:  Replace unixTime with slotIndex in Timelock/Expiration

### LINKED ISSUE #962: Replace unixTime with slotIndex in Timelock/Expiration - nodejs
https://github.com/iotaledger/tips-draft/blob/tip38/tips/TIP-0038/tip-0038.md#timelock-unlock-condition
https://github.com/iotaledger/tips-draft/blob/tip38/tips/TIP-0038/tip-0038.md#expiration-unlock-condition

### PR DESCRIPTION
**Pull request by kwek20**
Fixes #962

### CODE REVIEW

File: bindings/nodejs/lib/types/client/query-parameters.ts [hunk 1]
```diff
@@ -84,13 +86,13 @@ interface StorageDepositReturnAddress {
 interface HasTimelock {
     hasTimelock: boolean;
 }
-/** Return outputs that are timelocked before a certain Unix timestamp. */
+/** Return outputs that are timelocked before a certain Slot index. */
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/client/query-parameters.ts]: I wouldn't uppercase Slot like that

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Format

File: bindings/nodejs/lib/types/block/output/unlock-condition.ts [hunk 2]
```diff
@@ -124,41 +125,44 @@ class StorageDepositReturnUnlockCondition extends UnlockCondition /*implements I
...
-        this.unixTime = unixTime;
+        this.slotIndex = slotIndex;
     }
     /**
-     * Unix time (seconds since Unix epoch) starting from which the output can be consumed.
+     * Slot index starting from which the output can be consumed.
      */
-    getUnixTime(): number {
-        return this.unixTime;
+    getSlotIndex(): SlotIndex {
```
Comment (Thoralf-M) [inline bindings/nodejs/lib/types/block/output/unlock-condition.ts]: So here we are again, have a getter method or just make the field readable? :trollface:

Comment (kwek20) [reply bindings/nodejs/lib/types/block/output/unlock-condition.ts]: yea i saw it but thought maaaybe out of scope? :D ill update it all then

File: bindings/nodejs/examples/how_tos/advanced_transactions/advanced_transaction.ts [hunk 3]
```diff
@@ -36,9 +35,12 @@ async function run() {
         // To sign a transaction we need to unlock stronghold.
         await wallet.setStrongholdPassword(process.env.STRONGHOLD_PASSWORD);
 
+        const client = await wallet.getClient();
+
         // Create an ouput with amount 1_000_000 and a timelock of 1 hour
```
Comment (DaughterOfMars) [inline bindings/nodejs/examples/how_tos/advanced_transactions/advanced_transaction.ts]: ouput :S

File: bindings/nodejs/examples/how_tos/advanced_transactions/advanced_transaction.ts [hunk 4]
```diff
@@ -36,9 +35,12 @@ async function run() {
...
-        const in_an_hour = Math.floor(Date.now() / 1000) + 3600;
-        const basicOutput = await new Client({}).buildBasicOutput({
+        // TODO !!!
+        const slot_index = BigInt(1000); // client.getSlotIndex() + 1
```
Comment (DaughterOfMars) [inline bindings/nodejs/examples/how_tos/advanced_transactions/advanced_transaction.ts]: What's with the casing disparity?

File: bindings/nodejs/examples/how_tos/advanced_transactions/advanced_transaction.ts [hunk 5]
```diff
@@ -36,9 +35,12 @@ async function run() {
         // To sign a transaction we need to unlock stronghold.
         await wallet.setStrongholdPassword(process.env.STRONGHOLD_PASSWORD);
 
-        // Create an ouput with amount 1_000_000 and a timelock of 1 hour
-        const in_an_hour = Math.floor(Date.now() / 1000) + 3600;
-        const basicOutput = await new Client({}).buildBasicOutput({
+        const client = await wallet.getClient();
+
+        // Create an output with amount 1_000_000 and a timelock of 1 hour
+        // TODO !!!
```
Comment (DaughterOfMars) [inline bindings/nodejs/examples/how_tos/advanced_transactions/advanced_transaction.ts]: What is the TODO?

Comment (kwek20) [reply bindings/nodejs/examples/how_tos/advanced_transactions/advanced_transaction.ts]: get a proper timeslot, its the same todo as in thibaults pr. need some methods to calculate this

Comment (DaughterOfMars) [reply bindings/nodejs/examples/how_tos/advanced_transactions/advanced_transaction.ts]: Can you add that to the comment?

**Review by DaughterOfMars — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by abdulmth — APPROVED** [review]

**Pull request merged**